cns/ipampool/monitor.go

```diff
@@ -103,19 +103,20 @@ func (pm *Monitor) reconcile(ctx context.Context) error {
 	allocatedPodIPCount := len(pm.httpService.GetAllocatedIPConfigs())
 	pendingReleaseIPCount := len(pm.httpService.GetPendingReleaseIPConfigs())
 	availableIPConfigCount := len(pm.httpService.GetAvailableIPConfigs()) // TODO: add pending allocation count to real cns
-	requestedIPConfigCount := pm.spec.RequestedIPCount
+	requestedIPConfigCount := int(pm.spec.RequestedIPCount)
 	unallocatedIPConfigCount := cnsPodIPConfigCount - allocatedPodIPCount
-	freeIPConfigCount := requestedIPConfigCount - int64(allocatedPodIPCount)
+	currentFree := cnsPodIPConfigCount - allocatedPodIPCount
+	futureFree := requestedIPConfigCount - allocatedPodIPCount
```

Comment:
Can we rename it something else, futureFree is little confusing specially when it is being used for ScaleUP. You will scaleup if you dont have ips left. Cant think of a better name too :)

May be "ipsToBeAssignedToCNS"...I'm sure you will come with a better name

Reply:
I actually can't think of a better name...can we move forward with this one and I will follow up with a change to clean up the way this state is gathered and passed around?

Reply:
How About "totalIpsToBeAssignedToCNS"

Reply:
I think we're using assigned and allocated backwards. It should be: 
"IPs are allocated to CNS"
"CNS assigns IPs to Pods" 

Then, the total number of IPs that CNS _has_ are the its "allocated" IPs, and the total number if IPs _in use_ are how many that CNS has "assigned" to specific Pods.
"currentFree" becomes `unassigned := allocated - assigned`
"futureFree" becomes something like `requestedUnassigned := requested - assigned`

Reply:
I like the above analogy.
However, since we have defined allocated => CNS assigned ips to pod

lets rename currentFree = unAllocated
FutureFree = requestedUnAllocated

IMO, FutureFree is really hard to explain to a new pair of eyes.